ubcpi/ubcpi.py

```diff
@@ -0,0 +1,468 @@
+"""A Peer Instruction tool for edX by the University of British Columbia."""
+import os
+import random
+from copy import deepcopy
+import uuid
+
+from django.core.exceptions import PermissionDenied
+
+import pkg_resources
+from webob import Response
+
+from xblock.core import XBlock
+from xblock.exceptions import JsonHandlerError
+from xblock.fields import Scope, String, List, Dict, Integer, DateTime, Float
+from xblock.fragment import Fragment
+from answer_pool import offer_answer, validate_seeded_answers, get_other_answers
+import persistence as sas_api
+from serialize import parse_from_xml
+
+STATUS_NEW = 0
+STATUS_ANSWERED = 1
+STATUS_REVISED = 2
+
+# this is a limitation in the database submissions_submission.raw_answer field, which is a TEXT and has limit
+# of 64k in size. Because we are storing rationale and revised rationale both in the the field, the max size
+# for the rationale is half
+MAX_RATIONALE_SIZE = 32000
+
+
+# noinspection all
+class MissingDataFetcherMixin:
+    """ Copied from https://github.com/edx/edx-ora2/blob/master/openassessment/xblock/openassessmentblock.py """
```

Comment:
I have reservations about copying code from ORA2, because ORA2 is explicitly installed as a Django app, and  your PR to edx-platform did *not* install the PI block as a Django app. I'm concerned that code will unexpectedly fail because ORA2 is working under a different set of assumptions.